internal/db/db.go

```diff
@@ -35,11 +25,30 @@ func (db DbType) String() string {
 		"postgres",
 	}[db]
 }
+func StringToDbType(dialect string) (DbType, error) {
+	switch dialect {
+	case "postgres":
+		return Postgres, nil
+	default:
+		return UnknownDB, fmt.Errorf("%s is an unknown dialect", dialect)
+	}
+}
 
 // Open a database connection which is long-lived.
 // You need to call Close() on the returned gorm.DB
 func Open(dbType DbType, connectionUrl string) (*gorm.DB, error) {
```

Comment:
I think if there was a struct in the db package that was returned here instead of returning `*gorm.DB` directly, that would eliminate the need to import `gorm` throughout the rest of the code base, especially in a number of tests. Currently, in a number of tests it looks like `gorm` is import along with `db` since the result of `db.Open` is passed to other test helper functions. If instead we had something like:

```go
type DB {
  *gorm.DB
}

func Open(dbType DbType, connectionUrl string) (*DB, error) {
```
Then [this test](https://github.com/hashicorp/boundary/blob/23d2526e9a23c13958799de33cc0a9f1b46d3496/internal/db/schema/migrations/postgres/14/warehouse_user_dim_test.go#L143) for example would not need to import gorm.

Reply:
this is being addressed in #1535